include/faabric/util/snapshot.h

```diff
@@ -73,27 +76,95 @@ class SnapshotMergeRegion
 class SnapshotData
 {
   public:
-    size_t size = 0;
-    uint8_t* data = nullptr;
-    int fd = 0;
-
     SnapshotData() = default;
 
-    std::vector<SnapshotDiff> getDirtyPages();
+    explicit SnapshotData(size_t sizeIn);
+
+    SnapshotData(size_t sizeIn, size_t maxSizeIn);
+
+    explicit SnapshotData(std::span<const uint8_t> dataIn);
+
+    SnapshotData(std::span<const uint8_t> dataIn, size_t maxSizeIn);
+
+    SnapshotData(const SnapshotData&) = delete;
+
+    SnapshotData& operator=(const SnapshotData&) = delete;
+
+    ~SnapshotData();
+
+    void copyInData(std::span<const uint8_t> buffer, uint32_t offset = 0);
+
+    const uint8_t* getDataPtr(uint32_t offset = 0);
+
+    std::vector<uint8_t> getDataCopy();
+
+    std::vector<uint8_t> getDataCopy(uint32_t offset, size_t dataSize);
 
-    std::vector<SnapshotDiff> getChangeDiffs(const uint8_t* updated,
-                                             size_t updatedSize);
+    void mapToMemory(uint8_t* target);
 
     void addMergeRegion(uint32_t offset,
                         size_t length,
                         SnapshotDataType dataType,
                         SnapshotMergeOperation operation,
                         bool overwrite = false);
 
+    void clearMergeRegions();
+
+    std::map<uint32_t, SnapshotMergeRegion> getMergeRegions();
+
+    size_t getQueuedDiffsCount();
+
+    void queueDiffs(const std::vector<SnapshotDiff>& diffs);
```

Comment:
Accept any std::span instead of just std::vector (it's implicitly constructible, so this change allows more ways to call the function - like with a local stack-allocated array - without breaking the vector-based one)
```suggestion
    void queueDiffs(std::span<const SnapshotDiff> diffs);
```

Reply:
Nice, we have a lot of functions that take refs to vectors that can be replaced with this; will keep my eyes peeled 👀.